env/environment.go

```diff
@@ -197,6 +197,10 @@ func (e *Environment) Apply(diff Diff) {
 
 // Copy returns a copy of the env
 func (e *Environment) Copy() *Environment {
+	if e == nil {
+		return New()
+	}
+
```

Comment:
gotta catch 'em all ('em being nil pointer dereference panics)

Reply:
I sort of expect a copy of `nil` to be `nil`, but it's probably more pragmatic for it to be `New()`.